packages/wonder-blocks-tabs/src/components/navigation-tab-item.tsx

```diff
@@ -1,14 +1,38 @@
-import {addStyle} from "@khanacademy/wonder-blocks-core";
+import {addStyle, AriaProps, StyleType} from "@khanacademy/wonder-blocks-core";
 import {CSSProperties, StyleSheet} from "aphrodite";
 import * as React from "react";
 import {styles as typographyStyles} from "@khanacademy/wonder-blocks-typography";
 import {semanticColor, sizing} from "@khanacademy/wonder-blocks-tokens";
 
-type Props = {
+type NavigationTabItemLinkProps = {style: StyleType; "aria-current"?: "page"};
+
+type Props = AriaProps & {
+    /**
+     * The `Link` to render for the navigation tab item.
+     *
+     * When a `Link` component is passed in for the `children` prop,
+     * `NavigationTabItem` will inject props for the `Link`. For specific use
+     * cases where the `Link` component is wrapped by another component (like a
+     * `Tooltip` or `Popover`), a render function can be used instead. The
+     * render function provides the Link props that should be applied to the
+     * Link component. See example in the docs for more details.
+     */
+    children:
+        | React.ReactElement
+        | ((linkProps: NavigationTabItemLinkProps) => React.ReactElement);
+    /**
+     * An id for the root element.
+     */
+    id?: string;
     /**
-     * The contents of the NavigationTabItem (usually a Link component)
+     * Optional test ID for e2e testing.
      */
-    children: React.ReactElement;
+    testId?: string;
+    /**
+     * If the `NavigationTabItem` is the current page. If `true`, current
+     * styling and aria-current=page will be applied to the Link.
+     */
+    current?: boolean;
```

Comment:
Naming this the `current`  prop instead of `selected` since we'll use `aria-current` under the hood. This is also in sync with the Figma component! 

Note: In this PR, it's only used for semantics. Will use it for styling in the next PR!